How-To-Launch-SMART-on-FHIR-Apps.md

```diff
@@ -0,0 +1,65 @@
+REACT_APP_SMART_LAUNCH_URL = https://smartlauncher.interop.community/sample-app/launch?client_id=sampleapp&platform=meld
+```
+
+#### Where to grab the environment variable values
+
+1. `REACT_APP_CLIENT`: This is taken from Apps > Request Generator > Settings > Registered App Details > Client Id. Request Generator refers to the registered mcode/request-generator app in your Meld sandbox.
+2. `REACT_APP_EHR_BASE`, `REACT_APP_EHR_SERVER_TO_BE_SENT_TO_REMS_ADMIN_FOR_PREFETCH`, `REACT_APP_EHR_SERVER`: These are taken from your Meld sandbox's sidebar, under Settings > Sandbox > Secured FHIR Server URL.
+3. `REACT_APP_SMART_LAUNCH_URL`, `SMART_ENDPOINT`: This is taken from Apps > Sample App > Settings > Registered App Details > App Launch URI\*.
+
+## How to run
+
+1. Start request-generator normally and go to `http://localhost:3000/`. Click the "Authorize" button.
+2. Start rems-admin normally.
+3. In request-generator, click the "Select a Patient" button.
+4. Select Jon Snow (id: 130803).
+5. Click the "Launch SMART on FHIR app" button. This opens the SMART on FHIR app launch page provided as values to the `REACT_APP_SMART_LAUNCH_URL` and `SMART_ENDPOINT` environment variables.
+6. Click the "Authorize" button.
+7. You should see the expected SMART on FHIR app launch properly.
+8. Go back to request-generator and issue an order-sign hook, and click on the "Patient Enrollment Form" button.
+9. You should see the expected SMART on FHIR app launch.
+
+## Running other Meld-sandbox-registered SMART on FHIR Apps
+
+Log in to Meld at https://meld.interop.community/. Go to My Sandboxes > REMS > Apps to try out the other Registered Apps. The example above manually tests (1). You can try the remaining options after (2) just by changing the `REACT_APP_SMART_LAUNCH_URL` and `SMART_ENDPOINT` environment variables.
+
+1. Sample App - launches for (7) without needing to modify any of the supported CDS Hooks. You will get the following error at (9) unless you replace the `appContext` returned by the `src/hooks/rems.ordersign.ts` hook's `createSmartLink` function with a short string, e.g. `"hello"`.
+
+```
+HttpError: 500 Server Error
+URL: https://iol2auth.interop.community/token
+server_error: Exception [EclipseLink-4002] (Eclipse Persistence Services - 2.5.1.v20130918-f2b9fc5): org.eclipse.persistence.exceptions.DatabaseException
+Internal Exception: com.mysql.cj.jdbc.exceptions.MysqlDataTruncation: Data truncation: Data too long for column &#39;value&#39; at row 1
+Error Code: 1406
+Call: INSERT INTO launch_context (name, value) VALUES (?, ?)
+```
+
+This is just a bug with using Meld and not something we can fix; it is because the `appContext` string is too long.
+
+2. Bilirubin Risk Chart - launches for (7) and (9) without needing to modify any of the supported CDS Hooks. However, for (9) above, there will be a 500 error in the DevTools console, unless you modify the `appContext` as mentioned above.
```

Comment:
this system of referencing numbers in the list above is confusing.  I think it would be more clear if we just had it written out fully.

Reply:
I'm going to delete this section since it's too specific/not as general.